core/src/main/java/com/scalar/db/transaction/consensuscommit/RecoveryExecutor.java

```diff
@@ -272,4 +314,10 @@ public Result(
       this.recoveryFuture = recoveryFuture;
     }
   }
+
+  public enum RecoveryType {
```

Comment:
[minor] This can be package-private.
```suggestion
  enum RecoveryType {
```